lib/reqTemplate.js

```diff
@@ -75,42 +83,58 @@ function prepareTemplateString(template, defaultLookupPath) {
  * @returns {Function} a template resolver for the provided template
  */
 function compileTemplate(templateSpec, setValue, reqPart) {
-    var template;
-    var prevIndex = 0;
-    var completeTemplate = /^\{([^}]+)}$/.exec(templateSpec);
-    if (completeTemplate && completeTemplate.length > 0) {
-        template = [['raw', prepareTemplateString(completeTemplate[1], reqPart)]];
-    } else {
-        var re = /\{([^}]+)}/g;
-        template = [];
-        var match;
-        do {
-            match = re.exec(templateSpec);
-            if (match) {
-                if (match.index !== prevIndex) {
-                    template.push(templateSpec.substring(prevIndex, match.index));
-                }
-                template.push(['raw', prepareTemplateString(match[1], reqPart)]);
-                prevIndex = match.index + match[0].length;
-            } else if (prevIndex !== templateSpec.length) {
-                template.push(templateSpec.substring(prevIndex, templateSpec.length));
-            }
-        } while (match);
-    }
-
     var res;
+    var template = splitAndPrepareTAsseblyTemplate(templateSpec);
+    var callback = function(bit) {
+        if (res === undefined) {
+            res = bit;
+        } else {
+            res += '' + bit;
+        }
+    };
     var resolveTemplate = TAssembly.compile(template, {
+        nestedTemplate: true,
+        cb: callback
+    });
+    var options = {
         errorHandler: function() { return undefined; },
-        cb: function(bit) {
-            if (res === undefined) {
-                res = bit;
-            } else {
-                res += '' + bit;
+        cb: callback,
+        globals: {
+            default: function(val, defVal) {
+                return val || defVal;
+            },
+            merge: function(destination, source) {
```

Comment:
@d00rman It should, but I've just checked and it doesn't.. Let me handle it tomorrow morning